compass_app/app/lib/ui/search_form/widgets/search_form_continent.dart

```diff
@@ -45,117 +41,7 @@ class SearchFormContinent extends StatelessWidget {
-                return const SizedBox(width: 8);
-              },
-            );
-          },
-        ),
-      ),
-    );
-  }
-}
-
-class _CarouselItem extends StatelessWidget {
-  const _CarouselItem({
-    super.key,
-    required this.imageUrl,
-    required this.name,
-    required this.viewModel,
-  });
-
-  final String imageUrl;
-  final String name;
-  final SearchFormViewModel viewModel;
-
-  bool _selected() =>
-      viewModel.selectedContinent == null ||
-      viewModel.selectedContinent == name;
-
-  @override
-  Widget build(BuildContext context) {
-    return SizedBox(
-      width: 140,
-      height: 140,
-      child: ClipRRect(
-        borderRadius: BorderRadius.circular(10),
-        child: Stack(
-          children: [
-            CachedNetworkImage(
-              imageUrl: imageUrl,
-              fit: BoxFit.cover,
-              errorWidget: (context, url, error) {
-                // NOTE: Getting "invalid image data" error for some of the images
-                // e.g. https://rstr.in/google/tripedia/jlbgFDrSUVE
-                return const DecoratedBox(
-                  decoration: BoxDecoration(
-                    color: AppColors.grey3,
-                  ),
-                  child: SizedBox(width: 140, height: 140),
-                );
-              },
-            ),
-            Align(
-              alignment: Alignment.center,
-              child: Padding(
-                padding: const EdgeInsets.all(16.0),
-                child: Text(
-                  name,
-                  textAlign: TextAlign.center,
-                  style: GoogleFonts.openSans(
-                    fontSize: 18,
-                    fontWeight: FontWeight.w500,
-                    color: AppColors.white1,
-                  ),
-                ),
-              ),
-            ),
-            // Overlay when other continent is selected
-            Positioned.fill(
-              child: AnimatedOpacity(
-                opacity: _selected() ? 0 : 0.7,
-                duration: kThemeChangeDuration,
-                child: DecoratedBox(
-                  decoration: BoxDecoration(
-                    // Support dark-mode
-                    color: Theme.of(context).colorScheme.onPrimary,
-                  ),
-                ),
-              ),
-            ),
-            // Handle taps
-            Positioned.fill(
-              child: Material(
-                color: Colors.transparent,
-                child: InkWell(
-                  onTap: () {
-                    if (viewModel.selectedContinent == name) {
-                      viewModel.selectedContinent = null;
-                    } else {
-                      viewModel.selectedContinent = name;
-                    }
-                  },
-                ),
-              ),
-            ),
-          ],
-        ),
+        child: SearchFormContinentCarousel(viewModel: viewModel),
```

Comment:
I replaced this with a separate file ONLY because I'm writing about it in the documentation, so it's cleaner for the reader if the file is as small as possible. I might need to rename things to make this separation make sense.